### DISCUSSION TITLE
使用插件未生效

### DISCUSSION THREAD

**Discussion by yangxun521** [vuejs/pinia#2253] [Help and Questions]
![6EMI}XA TC94CP5}C@CRH~J](https://github.com/vuejs/pinia/assets/74896483/6bc9ff63-9b92-43e0-b623-e87f122aadc2)
![UD}4{FAFI ~PNO8C6VXTY9E](https://github.com/vuejs/pinia/assets/74896483/18c451ed-a3a7-45ea-b4f0-b3ef58378b13)
为什么独立封装成一个js，在main再调用 ，无法使持久化插件生效？
当我挪到main.js时，就没有问题了
而且，持久化存储不应该是常有的需求嘛？为什么不内置实现这个功能昂？还需要使用单独的插件依赖

**Comment by komine-jp** [comment]
我一个项目可以 另一个项目不可以

**Reply by yangxun521** [reply]
为什么没人回复我们.....

**Reply by komine-jp** [reply]
你在项目中引入了几次stores/index.js  如果其他位置有引入会导致失效   我这边换了写法  已经解决问题  https://www.cnblogs.com/cjc-0313/p/16970006.html

**Comment by Val-Juckett** [comment]
我也遇到了这个问题，令我困惑的是明明只是声明的位置不一样，怎么会导致这个问题，并不只是你提到的这个插件不生效，我自己编写的插件也遇到同样的问题
